### ISSUE TITLE
Problem With Incremental Compiler

### ISSUE THREAD

**Issue by jeffscottbrown** [micronaut-projects/micronaut-core#4277]
### Steps to Reproduce

1. Create new 2.1.0 app using `mn create-app demoapp` or Launch
2. `mn create-controller Demo`
3. Create any new java class below `src/main/java/`
4. Run `./gradlew classes`
5. Inspect `build/classes/java/main/META-INF/services/io.micronaut.inject.BeanDefinitionReference`, the `DemoController` bean definition will be there
6. Edit the java class created in step 3, any change at all, even just a comment
7. Run `./gradlew classes`

### Expected Behaviour

`build/classes/java/main/META-INF/services/io.micronaut.inject.BeanDefinitionReference` should exist and should include the `DemoController` bean definition.

### Actual Behaviour

`build/classes/java/main/META-INF/services/io.micronaut.inject.BeanDefinitionReference` does not exist.

### Additional Information

The way this manifested for me while working on a real app is I could run the app or run tests and everything is fine.  I made code changes to classes that aren't directly related to beans and then the app and tests start failing with bean not found errors because the services descriptor doesn't exist, or is missing some bean definitions.  Sometimes the file is there, but incomplete.

The problem does not happen if incremental compilation is disabled with something like this in `build.gradle`:

```
micronaut {
    runtime "netty"
    testRuntime "junit5"
    processing {
        incremental false
        annotations "demoapp.*"
    }
}
```

### Environment Information

- **Operating System**: macOs
- **Micronaut Version:** 2.1.0
- **JDK Version:** openjdk version "11.0.2" 2019-01-15

**Comment by jeffscottbrown** [comment]
The terminalizer session at [jeffscreencasts.s3.amazonaws.com/incrementalcompilationissue.gif](http://jeffscreencasts.s3.amazonaws.com/incrementalcompilationissue.gif) demonstrates what I am seeing.

**Comment by jeffscottbrown** [comment]
I did not do extensive testing around this, but it appears that the problem may not manifest if I use Gradle 6.6.  The bug report involves using the generated Gradle wrapper, which is Gradle `6.7-rc-3`.

**Comment by graemerocher** [comment]
A PR has been submitted to fix this in Gradle 6.7 https://github.com/gradle/gradle/pull/14813

Thanks @melix !
